File: DCP/1.js
// Given an array of numbers, return whether any two sums to K.
// For example, given [10, 15, 3, 7] and K of 17, return true since 10 + 7 is 17.

const two_sum = function(arr, k) {

  let valueToIndex = {};
  arr.map(function(element){
    valueToIndex[element] = 1;
  });

  for (let i = 0; i < arr.length; i++) {
    if ( (valueToIndex.hasOwnProperty(k - arr[i])) ) {
      return true;
    }
  }
  return false;
}

const test = two_sum([10, 15, 3, 7], 10);
console.log(test);

File: DCP/1.py
# Given an array of numbers, return whether any two sums to K.
# For example, given [10, 15, 3, 7] and K of 17, return true since 10 + 7 is 17.

def two_sum(lst, k):
  value_to_index = {}

  for i, num in enumerate(lst):
    value_to_index[num] = i

  print(value_to_index)
  for i, num in enumerate(lst):
    if (k - num) in value_to_index and i != value_to_index[k - num]:
      return true

  return false